File: app/services/provincia_service.py
from ..models import Provincia
from .. import db

def crear_provincia(data):
    nueva_provincia = Provincia(
        nombre=data["nombre"],
        pais=data["pais"],
        codigo=data["codigo"],
        tiene_municipio=data["tiene_municipio"]
    )
    db.session.add(nueva_provincia)
    db.session.commit()
    return nueva_provincia

def actualizar_provincia(id, data):
    provincia = Provincia.query.get(id)
    if not provincia:
        return None
    provincia.nombre = data.get("nombre", provincia.nombre)
    provincia.pais = data.get("puesto", provincia.pais)
    provincia.codigo = data.get("codigo", provincia.codigo)
    provincia.tiene_municipio = data.get("tiene_municipio", provincia.tiene_municipio)
    db.session.commit()
    return provincia

def eliminar_provincia(id):
    provincia = Provincia.query.get(id)
    if not provincia:
        return False
    db.session.delete(provincia)
    db.session.commit()
    return True

def obtener_provincias():
    provincias = Provincia.query.all()
    return [p.to_dict() for p in provincias]

File: Frontend/src/api/services/provinciaService.ts
import apiClient from '../client'
import type { Provincia, ProvinciaPayload } from '../interfaces'

export async function obtenerProvincias(): Promise<Provincia[]> {
  const { data } = await apiClient.get<Provincia[]>('/provincias')
  return data
}

export async function obtenerProvincia(id: number): Promise<Provincia> {
  const { data } = await apiClient.get<Provincia>(`/provincias/${id}`)
  return data
}

export async function crearProvincia(payload: ProvinciaPayload): Promise<Provincia> {
  const { data } = await apiClient.post<Provincia>('/provincias', payload)
  return data
}

export async function actualizarProvincia(id: number, payload: Partial<ProvinciaPayload>): Promise<Provincia> {
  const { data } = await apiClient.put<Provincia>(`/provincias/${id}`, payload)
  return data
}

export async function eliminarProvincia(id: number): Promise<{ mensaje: string }> {
  const { data } = await apiClient.delete<{ mensaje: string }>(`/provincias/${id}`)
  return data
}
